[Bug] InstantSearch component produces errors with typescript strict mode turned ON

## I'm submitting a...
* Bug report

## Current behavior
Algolia InstantSearch Angular component produces compile time errors with Angular CLI v6 on build with typescript strict mode turned ON

## Expected behavior
Algolia InstantSearch Angular component should successfully build with Angular CLI v6 when strict mode is turned `ON` in `tsconfig.json`.
If strict mode is turned OFF, the the build is successful.

## Minimal reproduction of the problem with instructions

```bash
> ng version
	- Angular CLI: 6.0.0-beta.4
> ng new instantsearch-test
```
- Now setup InstantSearch component as described here: https://community.algolia.com/angular-instantsearch/getting-started.html

- Enable strict mode in `tsconfig.json`

```bash
> yarn build (ng build -prod)
```

```typescript
ERROR in node_modules/angular-instantsearch/hits/hits.d.ts(3,22): error TS2415: Class 'NgAisHits' incorrectly extends base class 'BaseWidget'.
  Types of property 'updateState' are incompatible.
    Type '(state: { hits: {}[]; results: {}; }, isFirstRendering: boolean) => void' is not assignable to type '(state: {}, isFirstRendering: boolean) => void | Promise<void>'.
      Types of parameters 'state' and 'state' are incompatible.
        Type '{}' is not assignable to type '{ hits: {}[]; results: {}; }'.
          Property 'hits' is missing in type '{}'.
node_modules/angular-instantsearch/infinite-hits/infinite-hits.d.ts(2,22): error TS2415: Class 'NgAisInfiniteHits' incorrectly extends base class 'BaseWidget'.
  Types of property 'updateState' are incompatible.
    Type '(state: { hits: {}[]; results: {}; isLastPage: boolean; showMore: Function; }, isFirstRendering: ...' is not assignable to type '(state: {}, isFirstRendering: boolean) => void | Promise<void>'.
      Types of parameters 'state' and 'state' are incompatible.
        Type '{}' is not assignable to type '{ hits: {}[]; results: {}; isLastPage: boolean; showMore: Function; }'.
          Property 'hits' is missing in type '{}'.
node_modules/angular-instantsearch/range-slider/range-slider.d.ts(10,22): error TS2415: Class 'NgAisRangeSlider' incorrectly extends base class 'BaseWidget'.
  Types of property 'updateState' are incompatible.
    Type '(state: RangeSliderState, isFirstRendering: boolean) => void' is not assignable to type '(state: {}, isFirstRendering: boolean) => void | Promise<void>'.
      Types of parameters 'state' and 'state' are incompatible.
        Type '{}' is not assignable to type 'RangeSliderState'.
          Property 'range' is missing in type '{}'.
node_modules/angular-instantsearch/hierarchical-menu/hierarchical-menu-item.d.ts(10,14): error TS7006: Parameter 'string' implicitly has an 'any' type.
node_modules/angular-instantsearch/hierarchical-menu/hierarchical-menu-item.d.ts(11,17): error TS7006: Parameter 'string' implicitly has an 'any' type.
```


## Environment

Package versions:
```
npm ls --depth 0
+-- @angular/animations@6.0.0-beta.6
+-- @angular/cli@6.0.0-beta.4
+-- @angular/common@6.0.0-beta.6
+-- @angular/compiler@6.0.0-beta.6
+-- @angular/compiler-cli@6.0.0-beta.6
+-- @angular/core@6.0.0-beta.6
+-- @angular/forms@6.0.0-beta.6
+-- @angular/http@6.0.0-beta.6
+-- @angular/language-service@6.0.0-beta.6
+-- @angular/platform-browser@6.0.0-beta.6
+-- @angular/platform-browser-dynamic@6.0.0-beta.6
+-- @angular/router@6.0.0-beta.6
+-- @types/jasmine@2.8.6
+-- @types/jasminewd2@2.0.3
+-- @types/node@8.9.4
+-- angular-instantsearch@1.0.1
+-- codelyzer@4.1.0
+-- core-js@2.5.3
+-- html-webpack-plugin@2.30.1 extraneous
+-- jasmine-core@3.1.0
+-- jasmine-spec-reporter@4.2.1
+-- karma@2.0.0
+-- karma-chrome-launcher@2.2.0
+-- karma-coverage-istanbul-reporter@1.4.1
+-- karma-jasmine@1.1.1
+-- karma-jasmine-html-reporter@0.2.2
+-- protractor@5.3.0
+-- rxjs@5.5.6
+-- ts-node@5.0.0
+-- tslint@5.9.1
+-- typescript@2.6.2
`-- zone.js@0.8.20
```

Browser:
- Chrome (desktop) version 64.0.3282.186


System:
- TypeScript version: 2.6.2 (strict mode ON)
- Node version: 8.9.4
- Platform:  Windows 10 Pro 1709

--- a/src/hits/hits.ts
+++ b/src/hits/hits.ts
@@ -51,10 +51,7 @@ export class NgAisHits extends BaseWidget {
     this.createWidget(connectHits, { escapeHits: true });
   }
 
-  updateState = (
-    state: { hits: {}[]; results: {} },
-    isFirstRendering: boolean
-  ) => {
+  updateState = (state, isFirstRendering: boolean) => {
     if (isFirstRendering) return;
 
     this.state = {

--- a/src/infinite-hits/infinite-hits.ts
+++ b/src/infinite-hits/infinite-hits.ts
@@ -76,10 +76,7 @@ export class NgAisInfiniteHits extends BaseWidget {
     this.state.showMore();
   }
 
-  updateState = (
-    state: { hits: {}[]; results: {}; isLastPage: boolean; showMore: Function },
-    isFirstRendering: boolean
-  ) => {
+  updateState = (state, isFirstRendering: boolean) => {
     if (isFirstRendering) return;
 
     this.state = {

--- a/src/range-slider/range-slider.ts
+++ b/src/range-slider/range-slider.ts
@@ -69,7 +69,7 @@ export class NgAisRangeSlider extends BaseWidget {
     super.ngOnInit();
   }
 
-  public updateState = (state: RangeSliderState, isFirstRendering: boolean) => {
+  public updateState = (state, isFirstRendering: boolean) => {
     if (isFirstRendering) {
       // create slider
       const config = {

--- a/tsconfig-aot.json
+++ b/tsconfig-aot.json
@@ -13,6 +13,10 @@
     "sourceMap": true,
     "inlineSources": true,
     "strictNullChecks": true,
+    "noImplicitThis": true,
+    "alwaysStrict": true,
+    "strictFunctionTypes": true,
+    "strictPropertyInitialization": false,
     "noStrictGenericChecks": false,
     "skipLibCheck": true
   },

